src/widgets/widget_definitions.rs

```diff
@@ -268,6 +269,23 @@ fn build_gtk_expander(bargs: &mut BuilderArgs) -> Result<gtk::Expander> {
     Ok(gtk_widget)
 }
 
+/// @widget revealer extends container
+/// @desc A widget that can reveal a child with an animation.
+fn build_gtk_revealer(bargs: &mut BuilderArgs) -> Result<gtk::Revealer> {
+    let gtk_widget = gtk::Revealer::new();
+    resolve_block!(bargs, gtk_widget, {
+    // @prop transition - the name of the transition
+    prop(transition: as_string) { gtk_widget.set_transition_type(parse_transition(&transition)?); },
+    // @prop reveal - sets if the child is revealed or not
+    prop(reveal: as_bool) { gtk_widget.set_reveal_child(reveal); },
+    // @prop duration - the duration of the reveal animation
+    prop(duration: as_i32) {
+        gtk_widget.set_transition_duration(duration.abs() as u32);
+    },
```

Comment:
duration here should use the `util::parse_duration` format (so just rely on parse_duration here :P), otherwise this would be slightly confusing